### ISSUE TITLE
Not working with geyserMC

### ISSUE THREAD

**Issue by naanu88** [IzzelAliz/Arclight#1287]
When connecting with to server with mods on bedrock via geyser, says that you need to have forge installed.  Can't do that on bedrock.

**Comment by p0t4t0sandwich** [comment]
Yeah, that's cause you can't install Forge mods on Bedrock, it's never gonna work.
Use Paper/Spigot if you want to support Bedrock, that's the only way. Either that or with server-side-only mods, but at that point you'd be better off using Paper/Spigot.

**Comment by naanu88** [comment]
Ok

**Comment by Elyas2024** [comment]
i use fabric with serverside mods since spigot and its forks change some mechanics in minecraft. though arclight fabric exists for 1.21. why no support for geyser there?

**Comment by p0t4t0sandwich** [comment]
This post was referring to Forge.
Arclight Fabric + Geyser Fabric with serverside mods might work, if it doesn't, post your logs in a github issue.

**Comment by Elyas2024** [comment]
u could switch to either fabric or neoforge since geysermc do support them

**Comment by p0t4t0sandwich** [comment]
Not really the point though, since the user in the post wanted Forge content-mods to work with Bedrock (which is impossible lol)

**Comment by virtualspan** [comment]
This doesn't really matter, because you can just use Geyser-Standalone, which is a specialized geyser proxy, or geyser with velocity. Also if you're doing this, make sure you use the mod version of floodgate on the server from the [modrinth page](https://modrinth.com/mod/floodgate/versions), and use viaversion on the server, even if it doesn't work with java, it'll make Geyser work for some reason.

Edit: Of course you will still need to use only server-side mods, and there are instructions on how to setup Geyser-Standalone and Geyser with Velocity on the Geyser Setup page, linked below for convenience.
[Geyser-Standalone Wiki](https://geysermc.org/wiki/geyser/setup?platform=proxy-servers&host=self)
Geyser with Velocity Wiki: [Self-Host](https://geysermc.org/wiki/geyser/setup?platform=proxy-servers&host=self)  [Host Provider](https://geysermc.org/wiki/geyser/setup?platform=proxy-servers&host=provider)

**Comment by virtualspan** [comment]
Actual GeyserMC support would still be a plus though, because other methods do create more lag compared to just having geyser on the server.

**Comment by qyl27** [comment]
Is there any way to add new items to Bedrock Edition?  
As far as I know, haven't yet, in these use cases, Spigot might be a better choice.

**Comment by virtualspan** [comment]
@qyl27 Yeah, and it's possible with all versions of geyser, spigot or not. [There's an official wiki here](https://geysermc.org/wiki/geyser/custom-resource-packs) and there's an automatic tool for converting java resource packs to bedrock called [java2bedrock.sh](https://github.com/Kas-tle/java2bedrock.sh). For example, you can use something like Oxaren, ItemsAdder, MythicMobs, ModelEngine, or any plugins/mods that use custom resource packs and even just datapacks with required or optional resourcepacks, and convert the resource pack to bedrock, set it up with geyser and use custom items, mobs, decorations, etc with custom textures on bedrock.

**Comment by qyl27** [comment]
I believe there was no Forge/Fabric compatible way at least.

**Comment by p0t4t0sandwich** [comment]
They have a project in early development that aims to create a compatibility layer for modded items and such:
<https://geysermc.org/wiki/other/hydraulic>
Nothing concrete yet however

**Comment by virtualspan** [comment]
> I believe there was no Forge/Fabric compatible way at least.

Geyser has both Fabric and Neoforge versions, and everything mentioned with resource packs before works here too.
There is no Forge version however.

**Comment by p0t4t0sandwich** [comment]
Note: The main topic was adding modded content, not whether it worked on modloaders or not.

Edit: since you're limited to server-side mods, there's almost no point using a modloader over Paper (unless you're writing your own server-side utilities)

**Comment by qyl27** [comment]
> Note: The main topic was adding modded content, not whether it worked on modloaders or not.
> 
> Edit: since you're limited to server-side mods, there's almost no point using a modloader over Paper (unless you're writing your own server-side utilities)

I just want to talk about something on the existing mod loaders, as you said, I think we have no reason to run the Geyser for our modded server until it has a compat, it is no different between only server-side mods and Spigot + Geyser now.
